pandas/tests/indexes/test_base.py

```diff
@@ -305,6 +305,30 @@ def test_constructor_simple_new(self):
         result = idx._simple_new(idx, 'obj')
         tm.assert_index_equal(result, idx)
 
+    def test_constructor_names(self):
+        # test both `name` and `names` provided
+        with pytest.raises(TypeError):
+            idx = Index([1, 2, 3], name='a', names=('a',))
+
+        # test non-list-like `names`
+        with pytest.raises(TypeError):
+            idx = Index([1, 2, 3], names='a')
+
+        # test using `name` for a flat `Index`
+        idx = Index([1, 2, 3], name='a')
+        assert idx.name == 'a'
+        assert idx.names == ('a',)
+
+        # test using `names` for a flat `Index`
+        idx = Index([1, 2, 3], names=('a',))
+        assert idx.name == 'a'
+        assert idx.names == ('a',)
+
```

Comment:
I would personally replace the two blocks with a loop (over ``kwargs``, or over ``idx`` itself), but it's a matter of taste. You could also include the case ``name=('a',)``, which should still work.